src/Presentation/Presenters/CreditCardPaymentUrlGenerator.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Presentation\Presenters;
+
+use Symfony\Component\Translation\TranslatorInterface;
+use WMDE\Fundraising\Frontend\DonationContext\UseCases\AddDonation\AddDonationResponse;
+use WMDE\Fundraising\Frontend\PaymentContext\Domain\PaymentUrlGenerator\CreditCard;
+
+/**
+ * Render the credit card payment page embedding an iframe
+ *
+ * @licence GNU GPL v2+
+ */
+class CreditCardPaymentUrlGenerator {
+
+	private $translator;
+	private $urlGenerator;
+
+	public function __construct( TranslatorInterface $translator, CreditCard $urlGenerator ) {
+		$this->translator = $translator;
+		$this->urlGenerator = $urlGenerator;
+	}
+
+	public function buildUrl( AddDonationResponse $response ): string {
+		$personalInfo = $response->getDonation()->getDonor();
+
+		return $this->urlGenerator->generateUrl(
```

Comment:
They hid it well... Their payment notifier expects the notification handler to provide the return URL (just as we do it now). There is no other way to define this.

(see [example script](https://www.micropayment.de/redirector/?to=download&what=example-cc-evt))